sopel/lifecycle.py

```diff
@@ -13,21 +13,21 @@
 import inspect
 import logging
 import traceback
-from typing import Callable, Optional
+from typing import Callable
 
 from packaging.version import parse as parse_version
 
 from sopel import __version__
 
 
 def deprecated(
-    reason: Optional[str] = None,
-    version: Optional[str] = None,
-    removed_in: Optional[str] = None,
-    warning_in: Optional[str] = None,
+    reason: str | Callable | None = None,
```

Comment:
Should also update the doc for `reason` to mention that it can be a callable.

Reply:
This is a "magic" behavior that allows both `@deprecated` and `@deprecated()` to work.

I'm not convinced that it would be useful to explicitly document passing a `Callable` for this argument. It isn't meant to be done explicitly.

Reply:
Ah, right.